hnn_core/extracellular.py

```diff
@@ -0,0 +1,587 @@
+        ``positions``. Note that the size of `voltages` must be:
+        ``(n_trials, n_electrodes, n_times)``, i.e., three-dimensional.
+
+    Attributes
+    ----------
+    times : list of float
+        The time points the extracellular voltages are sampled at (ms)
+    voltages : list of list of list of float
+        A three-dimensional list with dimensions: (n_trials, n_electrodes,
+        n_times).
+    sfreq : float
+        Sampling rate of the extracellular data (Hz).
+
+    Notes
+    -----
+    See Table 5 in http://jn.physiology.org/content/104/6/3388.long for
+    measured values of conductivity in rat cortex (note units there are mS/cm)
+    """
+
+    def __init__(self, positions, *, conductivity=0.3, method='psa',
+                 min_distance=0.5, times=None, voltages=None):
+
+        _validate_type(positions, (tuple, list), 'positions')
+        if len(positions) == 3:  # a single coordinate given
+            positions = [positions]
+        for pos in positions:
+            _validate_type(pos, (tuple, list), 'positions')
+            if len(pos) != 3:
+                raise ValueError('positions should be provided as xyz '
+                                 f'coordinate triplets, got: {positions}')
+
+        _validate_type(conductivity, float, 'conductivity')
+        if not conductivity > 0.:
+            raise ValueError('conductivity must be a positive number')
+        _validate_type(min_distance, float, 'min_distance')
+        if not min_distance > 0.:
+            raise ValueError('min_distance must be a positive number')
+        if method is not None:  # method allowed to be None for testing
+            _validate_type(method, str, 'method')
+            _check_option('method', method, ['psa', 'lsa'])
+
+        if times is None:
+            times = list()
+        if voltages is None:
+            voltages = list()
+
+        _validate_type(times, list, 'times')
+        _validate_type(voltages, list, 'voltages')
+        for ii in range(len(voltages)):
+            if len(voltages[ii]) != len(positions):
+                raise ValueError(f'number of voltage traces must match number'
+                                 f' of channels, got {len(voltages[ii])} and '
+                                 f'{len(positions)} for trial {ii}')
+            for jj in range(len(voltages[ii])):
+                if len(times) != len(voltages[ii][jj]):
+                    raise ValueError('length of times and voltages must match,'
+                                     f' got {len(times)} and '
+                                     f'{len(voltages[ii][jj])} for trial{ii}, '
+                                     f'channel {jj}')
+
+        self.positions = positions
+        self.n_contacts = len(self.positions)
+        self.conductivity = conductivity
+        self.method = method
+        self.min_distance = min_distance
+
+        self._times = times
+        self._data = voltages
+
+    def __getitem__(self, trial_no):
```

Comment:
Quick line in the docstring detailing that slice indexing pulls out individual trials?